File: examples/typescript/remember-recall.ts
import { SoloClient } from "../../sdks/typescript/solo-client.js";

const solo = new SoloClient({
  baseUrl: process.env.SOLO_URL ?? "http://127.0.0.1:17821",
  bearerToken: process.env.SOLO_BEARER_TOKEN,
});

const content =
  process.argv.slice(2).join(" ") ||
  "Avery prefers planning notes with owners and dates.";

const saved = await solo.remember(content, {
  sourceType: "sdk_example",
  salience: 0.7,
});
console.log(`remembered ${saved.memory_id}`);

const recall = await solo.recall("planning notes owners dates", { limit: 3 });
for (const hit of recall.hits) {
  console.log(`- ${hit.memory_id}: ${hit.content}`);
}

const context = await solo.context("planning notes owners dates", {
  subject: "Avery",
  limit: 3,
});
console.log(
  `context: recall=${context.sections.recall.count} facts=${context.sections.facts.count} themes=${context.sections.themes.count}`,
);

File: examples/python/remember_recall.py
import os
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2] / "sdks" / "python"))

from solo_client import SoloClient  # noqa: E402

solo = SoloClient(
    base_url=os.getenv("SOLO_URL", "http://127.0.0.1:17821"),
    bearer_token=os.getenv("SOLO_BEARER_TOKEN"),
)
content = " ".join(sys.argv[1:]) or "Avery prefers planning notes with owners and dates."

saved = solo.remember(content, source_type="sdk_example", salience=0.7)
print(f"remembered {saved['memory_id']}")

for hit in solo.recall("planning notes owners dates", limit=3)["hits"]:
    print(f"- {hit['memory_id']}: {hit['content']}")

context = solo.context("planning notes owners dates", subject="Avery", limit=3)
sections = context["sections"]
print(
    "context: "
    f"recall={sections['recall']['count']} "
    f"facts={sections['facts']['count']} "
    f"themes={sections['themes']['count']}"
)
